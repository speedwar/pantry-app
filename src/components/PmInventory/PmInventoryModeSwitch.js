import React from 'react'
import { connect } from 'react-redux'
import { userContextActions } from 'rx/actions'

const PmInventoryModeSwitch = ({ dispatch, userContext }) => {
  const handleUserToggle = () => {
    const toggleUserType = userContext.userType === 'chef' ? 'storekeeper' : 'chef'
    const userTypeObj = { userType: toggleUserType }
    dispatch(userContextActions.updateUserType(userTypeObj))
  }
  
  return (
    <div className="l-grid l-grid--all-center p-2 m-3 border">
      <div className="l-grid__item">
        <span className="font-weight-bold">
          Switch to Mode:
        </span>
      </div>
      <div className="l-grid__item">
        <button
          className="btn btn-primary"
          onClick={ handleUserToggle }
        >
          {userContext.userType === 'chef' ? 'Storekeeper' : 'Chef'}
        </button>
      </div>
    </div>
  )
}

function mapStateToProps(state) {
  const {
    userContext
  } = state.userContextReducer
  return {
    userContext
  }
}

export default connect(mapStateToProps)(PmInventoryModeSwitch)
 