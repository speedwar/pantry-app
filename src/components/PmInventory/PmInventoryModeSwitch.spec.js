import React from 'react'
import { shallow } from 'enzyme'
import PmInventoryModeSwitch from './PmInventoryModeSwitch'
import { Provider } from 'react-redux'
import { store } from 'rx/store'

const defaultProps = {
  userContext: {
    userType: 'chef'
  },
}
const setup = (partialProps) => {
  const props = {
    ...defaultProps,
    ...partialProps,
  }
  const wrapper = shallow(
    <Provider store={ store } >
      <PmInventoryModeSwitch />
    </Provider>
  )
  return { wrapper, props }
}

describe('<PmInventoryModeSwitch /> component', () => {
  it('renders component with props', () => {
    const wrapper = setup({
      userContext: {
        userType: 'storekeeper'
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
