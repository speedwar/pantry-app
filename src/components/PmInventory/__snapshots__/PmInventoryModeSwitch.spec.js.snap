// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PmInventoryModeSwitch /> component renders component 1`] = `
Object {
  "props": Object {
    "userContext": Object {
      "userType": "storekeeper",
    },
  },
  "wrapper": <ContextProvider
    value={
      Object {
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "subscription": Subscription {
          "handleChangeWrapper": [Function],
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": null,
        },
      }
    }
  >
    <Connect(PmInventoryModeSwitch) />
  </ContextProvider>,
}
`;
